core/state/database.go

```diff
@@ -184,6 +236,7 @@ func (db *CachingDB) Reader(stateRoot common.Hash) (Reader, error) {
 			readers = append(readers, newFlatReader(snap))
 		}
 	}
+	ts := db.LoadTransitionState(stateRoot)
```

Comment:
It's very expensive before the Verkle activation;

The TransitionState won't be existent before the verkle migration, each lookup ends up with an non-existent database lookup, which traverses the entire LSM Tree layers.

Reply:
that's a fair point. and I see that your (now deleted) changes serve as a guard. But it lacks the ability to distinguish between the transition and post-transition phases. That might still save some hits though.

Reply:
https://github.com/ethereum/go-ethereum/pull/32084

As I suggested previously, we should expose `IsVerkle` information to the state, so that we can easily distinguish these scenarios respectively:

- Before the verkle activation: `IsVerkle == false`
- The first verkle block: `IsVerkle == true && rawdb.ReadTransitionState(stateRoot) == null`
- The blocks within the conversion:  `IsVerkle == true && rawdb.ReadTransitionState(stateRoot) != null && state.End == false`
- The blocks after the conversion: `IsVerkle == true && rawdb.ReadTransitionState(stateRoot) != null && state.End == true`

Reply:
The benefit is: we can skip the entire TransitionState checking before the verkle, it's the main blocker from merging this change.